### PR TITLE
Update KMS readme with vault quick start guide

### PR DESCRIPTION
**Pull request by poornas**
## Description

Adding a quick start guide to configure Vault for use as KMS with minio server
## Motivation and Context

Helps guide new users
## Regression
 No

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
We don't need a separate vault quick start guide, we should simply add it in the same documentation in the prerequisite section.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: docs/kms/README.md [hunk 1]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
...
+vault read auth/approle/role/my-role/role-id  # get Approle ID
+vault write -f auth/approle/role/my-role/secret-id
+
+```
+
+The AppRole ID, AppRole Secret Id, Vault endpoint and Vault key name can now be used to start minio server with Vault as KMS.
+
 ### 3. Environment variables
 
 You'll need the Vault endpoint, AppRole ID, AppRole SecretID, encryption key-ring name before starting Minio server with Vault as KMS
```
Comment (eco-minio) [inline docs/kms/README.md]: Should provide sample snippet with output that users can verify things are successfully working

File: docs/kms/README.md [hunk 2]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
 - AppRole based authentication with read/update policy for transit backend. In particular, read and update policy
   are required for the generate data key endpoint and decrypt key endpoint.
 
+Here is a sample quick start to configuring vault with a transit backend and Approle with correct policy 
+#### 2.1 Start vault server in dev mode 
+Note that this is insecure and stores all its data in memory,and not meant for production
```
Comment (eco-minio) [inline docs/kms/README.md]: Should clarify that running in -dev mode is what is insecure

File: docs/kms/README.md [hunk 3]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
 - AppRole based authentication with read/update policy for transit backend. In particular, read and update policy
```
Comment (eco-minio) [inline docs/kms/README.md]: Should include reference to Vault docs regarding transit

File: docs/kms/README.md [hunk 4]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
...
+  capabilities = [ "read", "update"]
+}
+
+EOF
+
+export VAULT_ADDR='http://127.0.0.1:8200'
+vault auth enable approle    # enable approle style auth
+vault secrets enable transit  # enable transit secrets engine
+vault write -f  transit/keys/my-minio-key  #define a encryption key-ring for the transit path
+vault policy write minio-policy ~/code/src/github.com/minio/vaultpolicy.hcl  #define a policy for AppRole to access transit path
```
Comment (eco-minio) [inline docs/kms/README.md]: Should amend to a shorter path to the sample file (from the heredoc it implies `pwd`?)

File: docs/kms/README.md [hunk 5]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
...
+vault write auth/approle/role/my-role token_num_uses=0  secret_id_num_uses=0  period=60s # period indicates it is renewable if tok renewed before the period is over
```
Comment (eco-minio) [inline docs/kms/README.md]: "...if token is renewed..."

File: docs/kms/README.md (hunk 1 above)
Comment (eco-minio) [reply docs/kms/README.md]: "...AppRole ID, AppRole SecretID and encryption key-ring name..."

File: docs/kms/README.md [hunk 6]
```diff
@@ -14,6 +14,43 @@ Vault as Key Management System requires following to be configured in Vault
 - AppRole based authentication with read/update policy for transit backend. In particular, read and update policy
   are required for the generate data key endpoint and decrypt key endpoint.
 
+Here is a sample quick start to configuring vault with a transit backend and Approle with correct policy
```
Comment (ebozduman) [inline docs/kms/README.md]: `start to configuring vault` => `start for configuring vault`
IMHO

**Review by ebozduman — DISMISSED** [review]

**Comment by poornas** [comment]
@eco-minio  && @ebozduman - updated PR.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by eco-minio — CHANGES_REQUESTED** [review]
>objects.This document explains how to configure Minio with Vault as KMS.
objects. This document explains how to configure Minio with Vault as KMS.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by poornas** [comment]
@ebozduman , addressed your feedback

File: docs/kms/README.md [hunk 7]
```diff
@@ -10,13 +10,50 @@ Install Minio - [Minio Quickstart Guide](https://docs.minio.io/docs/minio-quicks
 ### 2. Configure Vault
 Vault as Key Management System requires following to be configured in Vault
 
-- transit backend configured with a named encryption key-ring
-- AppRole based authentication with read/update policy for transit backend. In particular, read and update policy
-  are required for the generate data key endpoint and decrypt key endpoint.
+- [transit](https://www.vaultproject.io/docs/secrets/transit/index.html) backend configured with a named encryption key-ring
+- [AppRole](https://www.vaultproject.io/docs/auth/approle.html) based authentication with read/update policy for transit backend. In particular, read and update policy
+  are required for the Generate Data Key](https://www.vaultproject.io/api/secret/transit/index.html#generate-data-key) endpoint and [Decrypt Data](https://www.vaultproject.io/api/secret/transit/index.html#decrypt-data) endpoint.
```
Comment (ebozduman) [inline docs/kms/README.md]: Missing `[` in `...for the Generate Data Key](...`

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: docs/kms/README.md (hunk 7 above)
Comment (poornas) [reply docs/kms/README.md]: thanks,fixed.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by ebozduman — APPROVED** [review]
LGTM

**Comment by ebozduman** [comment]
I noticed there is travis failure after giving my LGTM.
I don't think the failure is related to this PR, but it is about:
```
--- FAIL: TestAPICopyObjectPartHandler (1.73s)
	object-handlers_test.go:1767: Test 1: FS: Expected the response status to be `200`, but instead found `500`
```

**Comment by poornas** [comment]
@ebozduman , travis passed on restart

**Review by eco-minio — APPROVED** [review]
LGTM

**Pull request merged**